src/components/Card.js

```diff
@@ -1,35 +1,41 @@
 class Card {
-  constructor({ data, handleImageClick }, cardSelector) {
-    this._data = data;
+  constructor(
+    { data, handleImageClick, handleDeleteButton, handleLikeButton },
+    cardSelector
+  ) {
     this._name = data.name;
     this._link = data.link;
+    this._id = data._id;
     this._cardSelctor = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteButton = handleDeleteButton;
+    this._handleLikeButton = handleLikeButton;
   }
 
   _setEventListeners() {
     this._cardElement
       .querySelector(".elements__delete-button")
       .addEventListener("click", () => {
-        this._handleDeleteButton();
+        this._handleDeleteButton(this);
       });
 
     this._cardElement
       .querySelector(".elements__like-button")
       .addEventListener("click", () => {
-        this._handleLikeButton();
+        this._handleLikeButton(this);
       });
 
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick(this._data);
     });
   }
 
-  _handleDeleteButton() {
+  removeCardElement() {
     this._cardElement.remove();
+    this._cardElement = null;
   }
 
-  _handleLikeButton() {
+  toggleLikeElement() {
```

Comment:
Needs correction: note, that now you should also update the isLiked state and render the likes button according to it

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```